lib/routes/me.js

```diff
@@ -204,16 +200,28 @@ module.exports = function(crowi, app) {
     })
   }
 
+  actions.authThirdParty = function(req, res) {
+    const { continue: continueUrl = '/' } = req.query
+
+    if (!config.crowi['auth:requireThirdPartyAuth'] || req.user.hasValidThirdPartyId()) {
+      req.session.callback = null
+      return res.redirect(continueUrl)
+    }
+    req.session.callback = '/me/auth/third-party'
+    return res.render('me/auth/third-party')
+  }
+
   actions.authGoogle = function(req, res) {
     const googleAuth = require('../util/googleAuth')(config)
     const { user: userData } = req
     const toDisconnect = !!req.body.disconnectGoogle
     const toConnect = !!req.body.connectGoogle
+    const callback = req.session.callback || '/me'
 
     if (toDisconnect) {
```

Comment:
Is it okay to disconnect from Google when `disablePasswordAuth` set `true`?
(GitHub auth as well)
Check the setting and validate here to prevent disconnect and be impossible to login.